# Customer Churn Prediction and Retention Strategy Analysis in Telecommunications

Customer churn directly impacts revenue stability. Retaining existing customers is typically 5–7 times cheaper than acquiring new ones. This project aims to identify key churn drivers and build predictive models to support proactive retention strategies.

## Data Cleaning and Exploratory Data Analysis

This section focuses on:

* Assessing data quality and structure

* Cleaning and preparing the dataset for analysis

* Exploring customer behavior patterns

* Identifying potential drivers of churn

Through exploratory data analysis (EDA), we aim to uncover meaningful insights that will guide feature selection and model development in later stages.

### Data Cleaning

The following checks were performed using Python:
- Missing values
- Duplicate records
- Data types
- Invalid or unrealistic values

These checks help ensure that the analysis is based on reliable and accurate data.



In [1]:
import pandas as pd

df = pd.read_csv('../data/Telco_Cusomer_Churn.csv') # Loading data
df.head(5)

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


#### 1. Inspecting Dataset Structure

In [7]:
print("Dataset Size: ", df.shape)
df.info()

Dataset Size:  (7043, 21)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling

"TotalCharges" is stored as a Object type, however, it should be stored as a numerical type

In [9]:
df['TotalCharges'] = pd.to_numeric(df["TotalCharges"], errors="coerce")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


#### 2. Checking for Missing Values

In [8]:
df.isna().sum()

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64

No missing values 

#### 3. Checking for Duplicates


In [10]:
df.duplicated().sum()

0

No duplicates in the dataset

#### 4. Checking for Invalid or Unrealistic Values

In [11]:
df.describe()

,SeniorCitizen,tenure,MonthlyCharges,TotalCharges
count,7043.000000,7043.000000,7043.000000,7032.000000
mean,0.162147,32.371149,64.761692,2283.300441
std,0.368612,24.559481,30.090047,2266.771362
min,0.000000,0.000000,18.250000,18.800000
25%,0.000000,9.000000,35.500000,401.450000
50%,0.000000,29.000000,70.350000,1397.475000
75%,0.000000,55.000000,89.850000,3794.737500
max,1.000000,72.000000,118.750000,8684.800000


The dataset does not seem to have invalid or unrealistic values.

#### Data Cleaning Summary

The dataset was found to be clean and well-structured. No missing values or duplicate records were detected. 
All variables were stored using appropriate data types, and no invalid or unrealistic values were observed 
in key numerical fields such as price, quantity, and total sales. As a result, no additional cleaning steps 
were required before proceeding to exploratory data analysis.




---

## Exploratory Data Analysis (EDA)

Exploratory Data Analysis (EDA) was conducted using Tableau to visually examine patterns, relationships, and trends associated with customer churn. The objective was to identify high-risk customer segments and uncover behavioral and financial factors contributing to churn.

#### 1. Churn Overview Dashboard

The Churn Overview Dashboard provides a high-level summary of customer churn patterns within the telecommunications dataset. The goal of this dashboard is to quantify the overall churn rate and identify whether key demographic segments contribute disproportionately to customer attrition.

Dashboard Objective

Answer these core questions:

1. What is the overall churn rate?
2. How many customers have churned?
3. Are certain demographic groups more likely to churn?
4. Is churn evenly distributed across customer segments?


Key Findings:

Exploratory analysis revealed the following high-level churn metrics:

Total Customers: 7,000

Total Churned Customers: 1,869

Churn Rate: 26.54%

Interpretation:

Approximately 1 in 4 customers (26.54%) have churned.

This indicates a significant retention challenge, as more than a quarter of the customer base has discontinued service. In the telecommunications industry, where customer acquisition costs are typically high, this churn rate represents substantial potential revenue loss.

#### 2. Contract & Tenure Analysis

Contract type and customer tenure are critical factors in understanding churn behavior. Long-term customer commitment and relationship duration often reflect customer satisfaction, perceived value, and switching costs.

This section analyzes how contract structure and length of service influence customer churn.

The output from Tableau is presented below:

<img src="../Visualisations/Contract_and_Tenure.png" width="1200" height="150" />

Key Finding:

| Contract Type  | Churn Rate |
| -------------- | ---------- |
| Month-to-month | 42.71%     |
| One-year       | 11.27%     |
| Two-year       | 2.83%      |

Interpretation:

* Customers on month-to-month contracts are at the highest risk of churn, with over 42% leaving the service.

* Customers with one-year contracts have a much lower churn rate (~11%).

* Customers on two-year contracts exhibit minimal churn (~3%), showing strong loyalty.

Business Insight:

Customers on month-to-month contracts with short tenure represent the highest-risk segment, combining low contractual commitment with limited loyalty. These customers should be the primary focus of retention strategies.

#### 3. Revenue & Charges Analysis

Customer billing patterns, including monthly charges and total revenue, often influence churn behavior. Customers who pay more or perceive poor value may be more likely to leave, while high-value customers represent a bigger revenue risk if they churn.

This section analyzes how monthly charges and total charges relate to churn, highlighting financial patterns and potential risk segments.


##### Monthly Charges vs Churn

The output from Tableau is given below representing the Monthly Charges vs Churn:

<img src="../Visualisations/Revevnue_and_Charge.png" width="600" height="600" />


Key Findings:

| Monthly Charges ($) | Churn Rate |
| ------------------- | ---------- |
| 10                  | 8.97%      |
| 20                  | 10.29%     |
| 30                  | 27.72%     |
| 40                  | 32.39%     |
| 50                  | 20.88%     |
| 60                  | 19.45%     |
| 70                  | 39.82%     |
| 80                  | 36.21%     |
| 90                  | 37.80%     |
| 100                 | 32.95%     |
| 110                 | 13.70%     |

Interpretation:

1. High churn at mid-to-high charges:

    * Customers paying $30–$100/month have the highest churn rates, peaking around $70–$90.

    * This indicates that higher monthly charges can increase churn risk, possibly due to perceived cost vs. value.

2. Low churn at very low and very high charges:

    * Customers at the lowest end ($10–$20) or highest end ($110) show lower churn.

    * Low-charging customers may have minimal expectations, while the very high end may include long-tenure or premium users with higher loyalty.

3. Non-linear relationship:

    * Churn doesn’t increase perfectly with monthly charges — it rises in the middle range and drops slightly at extremes.

Possible Insights:

1. Target mid-range customers for retention:
Customers paying $30–$90/month are most likely to churn. Retention programs such as discounts, loyalty perks, or value-added services may reduce attrition.

2. High-value customers require protection:
Even though the highest payers ($110) have lower churn, losing them would have significant revenue impact. Specialized attention or VIP programs may be warranted.

3. Low-charging customers:
Minimal churn risk, but revenue contribution is low, retention efforts here are less urgent.

##### Total Charges vs Churn

Total charges represent the total amount a customer has paid over their lifetime with the company. This variable is closely related to customer tenure, as customers who remain longer naturally accumulate higher total charges.

The output from Tableau is given below representing the Total charges vs Churn:

<img src="../Visualisations/Total Charges vs Churn.png" width="600" height="600" />


Key Findings:

The churn rate varies significantly across different total charge levels, as shown below:

| Total Charges Range | Churn Rate |
| ------------------- | ---------- |
| < 500               | 41.45%     |
| 500                 | 26.99%     |
| 1000                | 22.27%     |
| 1500                | 19.03%     |
| 2000                | 28.41%     |
| 2500                | 26.62%     |
| 3000                | 23.13%     |
| 3500                | 14.89%     |
| 4000                | 17.86%     |
| 4500                | 15.71%     |
| 5000                | 17.65%     |
| 5500                | 13.81%     |
| 6000                | 15.98%     |
| 6500                | 14.04%     |
| 7000                | 12.78%     |
| 7500                | 11.21%     |
| 8000                | 2.82%      |
| 8500                | 12.50%     |

A clear pattern emerges from the analysis:

* Customers with very low total charges (<500) exhibit the highest churn rate (41.45%).

* As total charges increase, the churn rate generally declines.

* Customers with very high lifetime spending (above 7000) show very low churn rates, indicating strong loyalty.

The extremely low churn at 8000 total charges (2.82%) further suggests that long-term customers rarely leave the service.

Although there are minor fluctuations between some ranges, the overall trend indicates decreasing churn with increasing total charges.



Business Insight:

* Customers with low total charges are typically new customers, meaning they have had limited time to build loyalty with the company.

* Customers with high total charges are long-term customers, making them less likely to switch providers.

This suggests that the early stage of the customer lifecycle represents the highest churn risk.


Strategic Implications:

The company should prioritize retention strategies for new customers, including:

* Improved onboarding experiences

* Early engagement programs

* Targeted promotions or loyalty incentives during the first year of service

By reducing churn among these early-stage customers, the company could significantly improve long-term retention and revenue stability.

### 4. Services & Subscription Behavior

Customer subscription choices can significantly influence churn behavior. Value-added services such as technical support, security features, and backup services may increase customer satisfaction and reduce the likelihood of switching providers.

This section examines the relationship between service subscriptions and customer churn, with the goal of identifying services that contribute to stronger customer retention.

##### a) Internet Service Type vs Customer Churn

Goal:
* Understand how different internet service types relate to customer churn.
* This helps determine whether certain service offerings are associated with higher churn risk.

The output from Tableau is presented below:

<img src="../Visualisations/Internet Service Type vs Customer Churn.png" width="200" height="600" />

**Key insights**:

| Internet Service    | Churn Rate |
| ------------------- | ---------- |
| DSL                 | **18.96%** |
| Fiber Optic         | **41.89%** |
| No Internet Service | **7.40%**  |


1. Fiber Optic Customers Have the Highest Churn

Customers using fiber optic internet have a churn rate of 41.9%, which is more than double that of DSL customers and nearly six times higher than customers without internet service.

This suggests that fiber optic customers represent the highest-risk customer segment.

Possible reasons include:

* Higher service pricing

* Greater competition in high-speed internet markets

* Higher expectations for performance and reliability

2. DSL Customers Show Moderate Churn

DSL users have a churn rate of 18.9%, which is significantly lower than fiber optic but still notable.

This could indicate that DSL customers are relatively stable but may still be influenced by pricing or service quality.

3. Customers Without Internet Service Rarely Churn

Customers without internet service have a churn rate of 7.4%, the lowest among all groups.

This likely reflects:

* Lower monthly costs

* Simpler service packages

* Fewer competitive alternatives



**Business Implications**

The company should prioritize customer retention strategies for fiber optic users, since they represent the most volatile segment.

Potential strategies include:

* Improving fiber service reliability

* Offering bundled services (tech support, security, streaming)

* Providing loyalty discounts for long-term fiber customers

##### b) Tech Support vs Churn

Goal: 
* Analyze whether customers with technical support subscriptions are less likely to churn.
* Tech support is a value-added service, and it may increase customer satisfaction and reduce frustration when issues occur.

The results from Tableau are presented below:

<img src="../Visualisations/Tech Support vs Churn.png" width="200" height="600" />


**Key Insights:**

| Tech Support        | Churn Rate |
| ------------------- | ---------- |
| No                  | **41.64%** |
| Yes                 | **15.17%** |
| No Internet Service | **7.40%**  |

1 *Customers Without Tech Support Have the Highest Churn*

* Customers without technical support have a churn rate of 41.6%, which is nearly three times higher than customers who subscribe to tech support.

* This suggests that customers without access to assistance may experience unresolved technical issues, leading to dissatisfaction and ultimately churn.


2 *Tech Support Significantly Reduces Churn*

   Customers who subscribe to tech support show a much lower churn rate of 15.2%.

   This indicates that support services may improve the overall customer experience by:

   * Resolving service issues faster

   * Reducing frustration with internet or telecom problems

   * Increasing customer confidence in the provider

3 *Customers Without Internet Service Rarely Churn*

Customers with no internet service show the lowest churn rate (7.4%), consistent with earlier findings. These customers likely have simpler service packages and fewer alternatives in the market.


**Business Implications**

The results strongly suggest that technical support is an effective retention driver.

The company could reduce churn by:

* Bundling Tech Support with internet packages

* Offering discounted support services for new customers

* Targeting high-risk customers without support with promotional offers

Encouraging more customers to subscribe to tech support may significantly improve retention.

##### c) Online Security vs Customer Churn

**Goal:**

* Analyze whether customers who subscribe to online security services are less likely to churn.

* Online security is typically offered as an add-on service, and customers who adopt it may be more engaged with the provider.

<img src="../Visualisations/Online Security vs Churn.png" width="200" height="600" />


**Key Insights**

| Online Security     | Churn Rate |
| ------------------- | ---------- |
| No                  | **41.77%** |
| Yes                 | **14.61%** |
| No Internet Service | **7.40%**  |

1, *Customers Without Online Security Churn the Most*

Customers without online security have a churn rate of 41.8%, almost three times higher than those who subscribe to the service.

This shows that lack of value-added security services is strongly associated with higher churn.


2, *Online Security Reduces Churn Significantly*

Customers with online security subscriptions churn at only 14.6%, indicating that the service contributes to customer retention and loyalty.

Likely reasons include:

* Customers feel their digital assets are protected
* They are more engaged with the provider
* They perceive higher overall value from the subscription

3 *Customers Without Internet Service Rarely Churn*

As before, customers with no internet service show the lowest churn rate (7.4%), reflecting simple, low-cost service packages with minimal alternatives.


**Business Implications**

* Promote Online Security: Bundling online security with other services could significantly reduce churn.
* Target High-Risk Customers: Customers without security services are most likely to leave, so special offers or incentives could retain them.
* Enhance Value Perception: Services that improve trust and security are effective retention tools.

##### d) Streaming Services vs Customer Churn

**Goal:**

* Analyze how subscriptions to Streaming TV and Streaming Movies relate to churn.
* These are add-on entertainment services, and customers who use them may be slightly more engaged, but the effect on churn is usually smaller than for support/security services.

<img src="../Visualisations/Streaming Services vs Customer Churn.png" width="200" height="600" />

**Key Insights**


| Streaming TV        | Churn Rate |
| ------------------- | ---------- |
| No                  | **33.52%** |
| Yes                 | **30.07%** |
| No Internet Service | **7.40%**  |


1, *Customers With Streaming TV Have Slightly Lower Churn:*

Interestingly, customers who subscribe to streaming TV churn slightly less (30.1%) than those who don’t (33.5%).

This suggests that streaming TV may help slightly with engagement, but the effect is not as strong as Tech Support or Online Security.

2, *Customers Without Internet Service Have the Lowest Churn:*

Consistently, customers with no internet service show the lowest churn (7.4%), as they tend to have simple, low-cost packages with fewer alternatives.

**Business Implications**

* Streaming TV is not a major churn reducer, unlike Tech Support or Online Security.
* It may slightly improve engagement, but churn remains relatively high for internet-using customers.
* Focus on service reliability and bundling with value-added services to retain these users.

# Conclusion: Customer Churn Analysis

The exploratory data analysis (EDA) of the telecommunications dataset provides several **clear insights into customer churn behavior**, highlighting patterns related to **contracts, tenure, revenue, and service subscriptions**.


### 1. Overall Churn Overview

* Out of 7,043 customers, **1,869 churned**, giving an overall churn rate of **26.5%**.
* Churn is concentrated among **shorter-tenure customers** and those with **lower total charges**, indicating that **newer customers are the highest-risk group**.



### 2. Contract & Tenure Analysis

* **Month-to-month contracts** have the highest churn rate (**42.7%**), while **one-year** and **two-year contracts** have significantly lower churn (**11.3% and 2.8%**, respectively).
* Longer contracts provide **stability and reduced churn**, emphasizing the importance of incentivizing customers toward longer-term plans.


### 3. Revenue & Charges Analysis

* **Monthly Charges:** Customers with higher monthly bills generally exhibit higher churn, though the trend is non-linear.
* **Total Charges:** Customers with **low total charges (<500)** have the highest churn (**41.5%**), while high-spending long-term customers show very low churn (**2.8–12%**).
* These findings confirm that **early-stage customers and lower tenure users are at the highest risk of leaving**.


### 4. Services & Subscription Behavior

* **Internet Service:** Fiber optic users churn the most (**41.9%**) compared to DSL (**18.9%**) and no internet (**7.4%**).
* **Tech Support:** Customers without tech support churn at **41.6%**, while those with support churn only **15.2%**.
* **Online Security:** Lack of security services is associated with high churn (**41.8%**) versus **14.6%** for subscribers.
* **Streaming Services:** Minor effect; streaming TV subscribers churn slightly less (**30.1%**) than non-subscribers (**33.5%**).

**Key Insight:** Value-added services like Tech Support and Online Security are **the strongest retention drivers**, while entertainment services have only a modest effect.


### 5. Actionable Recommendations

1. **Focus on Early-Stage Customers:** Introduce onboarding, engagement, and loyalty programs to reduce churn among new customers.
2. **Promote Long-Term Contracts:** Encourage month-to-month customers to switch to one- or two-year contracts through incentives.
3. **Highlight Value-Added Services:** Offer Tech Support and Online Security bundles, especially to high-risk fiber-optic and new customers.
4. **Monitor High-Paying Customers:** Address service quality concerns for high-monthly-bill or fiber-optic users to prevent preventable churn.


### Overall Summary

The analysis reveals that **customer churn is heavily influenced by contract type, tenure, and value-added service subscriptions**. Early intervention strategies targeting **new and high-risk customers**, combined with **promoting support and security services**, can significantly reduce churn and increase long-term revenue.

This EDA provides a **strong foundation for predictive modeling** to identify at-risk customers and implement **targeted retention strategies**.
